packages/expo-cli/src/commands/build-native/iOSBuilder.ts

```diff
@@ -0,0 +1,124 @@
+import { ensureCredentialsAsync } from './credentials';
+import { Builder, BuilderContext } from './build';
+import {
+  CredentialsSource,
+  Workflow,
+  iOSBuildProfile,
+  iOSGenericBuildProfile,
+  iOSManagedBuildProfile,
+} from '../../easJson';
+
+interface CommonJobProperties {
+  platform: Platform.iOS;
+  projectUrl: string;
+  secrets?: {
+    provisioningProfileBase64: string;
+    distributionCertificate: {
+      dataBase64: string;
+      password: string;
+    };
+  };
+}
+
+class iOSBuilder implements Builder {
+  private credentials?: iOSCredentials;
+  private buildProfile: iOSBuildProfile;
+
+  constructor(public readonly ctx: BuilderContext) {
+    if (!ctx.eas.builds.ios) {
+      throw new Error("missing ios configuration, shouldn't happen");
+    }
+    this.buildProfile = ctx.eas.builds.ios;
+  }
+
+  public async prepareJobAsync(archiveUrl: string): Promise<Job> {
+    if (this.buildProfile.workflow === Workflow.Generic) {
+      return sanitizeJob(await this.prepareGenericJobAsync(archiveUrl, this.buildProfile));
+    } else if (this.buildProfile.workflow === Workflow.Managed) {
+      return sanitizeJob(await this.prepareManagedJobAsync(archiveUrl, this.buildProfile));
+    } else {
+      throw new Error("Unknown workflow. Shouldn't happen");
+    }
+  }
+
+  public async ensureCredentialsAsync(): Promise<void> {
+    if (!this.shouldLoadCredentials()) {
+      return;
+    }
+    const bundleIdentifier = this.ctx.exp?.ios?.bundleIdentifier;
+    if (!bundleIdentifier) {
+      throw new Error('"expo.ios.bundleIdentifier" field is required in your app.json');
+    }
+    const provider = new iOSCredentialsProvider(this.ctx.projectDir, {
+      projectName: this.ctx.projectName,
+      accountName: this.ctx.accountName,
+      bundleIdentifier,
+    });
+    await provider.initAsync();
+    const credentialsSource = await ensureCredentialsAsync(
+      provider,
+      this.buildProfile.workflow,
+      this.buildProfile.credentialsSource
+    );
+    this.credentials = await provider.getCredentialsAsync(credentialsSource);
+  }
+
+  private async prepareJobCommonAsync(archiveUrl: string): Promise<Partial<CommonJobProperties>> {
+    const secrets = this.credentials
+      ? {
+          secrets: {
+            provisioningProfileBase64: this.credentials.provisioningProfile,
+            distributionCertificate: {
+              dataBase64: this.credentials.distributionCertificate.certP12,
+              password: this.credentials.distributionCertificate.certPassword,
+            },
+          },
+        }
+      : {};
+
+    return {
+      platform: Platform.iOS,
+      projectUrl: archiveUrl,
+      ...secrets,
+    };
+  }
+
+  private async prepareGenericJobAsync(
+    archiveUrl: string,
+    buildProfile: iOSGenericBuildProfile
+  ): Promise<Partial<iOS.GenericJob>> {
+    return {
+      ...(await this.prepareJobCommonAsync(archiveUrl)),
+      type: BuildType.Generic,
```

Comment:
This line is causing an error for me when running `expo eas:build -p ios` (tried both `expo-cli@3.28.2` and `expo-cli@3.28.3`). BuildType is not defined.

Screenshot (from `expo-cli@3.28.2`):
<img width="100%" alt="Screen Shot 2020-11-03 at 09 23 14" src="https://user-images.githubusercontent.com/369384/98023732-37134900-1dbc-11eb-92f6-7c78606fd848.png">

`BuildType` is imported from `@expo/build-tools` and I checked on my system `/usr/local/lib/node_modules/expo-cli/node_modules/@expo/eas-build-job` is at v0.1.2.

 I was able to workaround by manually editing `/usr/local/lib/node_modules/expo-cli/build/commands/eas-build/build/builders/iOSBuilder.js` and changing this line to `type: 'generic',` but just wanted to report this somewhere.